src/crimson/os/seastore/transaction.h

```diff
@@ -539,6 +547,8 @@ class Transaction {
   /// Extents with pre-allocated addresses,
   /// will be added to written_ool_block_list after write
   std::list<LogicalCachedExtentRef> pre_alloc_list;
+  /// will be added to mutated_ool_block_list at submit_transaction phase
+  std::list<LogicalCachedExtentRef> inplace_delta_overwrite_list;
```

Comment:
Same, looks this list is unnecessary, and `pre_alloc_list` can be reused instead ?

Reply:
The reason why I separated this list is to avoid calling free blocks when the transaction conflicts with others---the extents in `pre_alloc_list` would release the allocated blocks when the conflict occurs.